configure

```diff
@@ -446,6 +446,18 @@ parser.add_option('--with-external-spidermonkey-release',
     default='',
     help='enable linking against an out of tree SpiderMonkey release build')
 
+parser.add_option('--with-system-nspr',
+    action='store',
+    dest='system_nspr',
+    default='',
+    help='use system NSPR when linking against an out-of-tree SpiderMonkey')
```

Comment:
The meaning of "system" isn't very clear to me.  What these flags are actually controlling is whether you need to have a copy of the NSPR/NSS shared libraries as a runtime dependency.  For example, if you have an out of tree Gecko build, you will need these dependencies.  If you have an external SpiderMonkey build, you won't.

Maybe `--external-spidermonkey-has-nspr` is a better name to use?